File: frontend/src/services/notificationService.js
import api from './api';

export const notificationService = {
    getMyNotifications: () => {
        return api.get('/notifications');
    },

    getUnreadCount: () => {
        return api.get('/notifications/unread-count');
    },

    markAsRead: (id) => {
        return api.put(`/notifications/${id}/read`);
    }
};

File: backend/src/main/java/com/ems/service/NotificationService.java
package com.ems.service;

import com.ems.dto.response.NotificationResponse;

import java.util.List;

public interface NotificationService {
    void createGlobalNotification(String message);
    void createTargetedNotification(Long employeeId, String message);
    List<NotificationResponse> getMyNotifications(Long employeeId);
    Long getUnreadCount(Long employeeId);
    void markAsRead(Long notificationId, Long employeeId);
}
